src/app/[locale]/page.tsx

```diff
@@ -2,97 +2,23 @@ import datasetFetcher from '@/lib/dataset-fetcher-instance';
 import {useLocale, NextIntlClientProvider} from 'next-intl';
 import {getTranslations} from 'next-intl/server';
 import ClientFilters from './client-filters';
-import DatasetList, {Sort, defaultSort} from './dataset-list';
-import {
-  SearchOptions,
-  SearchResult,
-  SortBy,
-  SortOrder,
-} from '@/lib/dataset-fetcher';
-
-const sortMapping = {
-  [Sort.RelevanceDesc]: {
-    sortBy: SortBy.Relevance,
-    sortOrder: SortOrder.Descending,
-  },
-  [Sort.NameAsc]: {
-    sortBy: SortBy.Name,
-    sortOrder: SortOrder.Ascending,
-  },
-  [Sort.NameDesc]: {
-    sortBy: SortBy.Name,
-    sortOrder: SortOrder.Descending,
-  },
-};
+import DatasetList from './dataset-list';
+import {fromSearchParamsToSearchOptions} from '@/lib/search-params';
 
 interface Props {
-  searchParams?: {
-    publishers?: string;
-    licenses?: string;
-    query?: string;
-    offset?: string;
-    sort?: Sort;
-  };
-}
-
-interface ErrorResponse {
-  hasError: true;
-  errorMessage: string;
-  searchResult: null;
+  searchParams?: object;
 }
-interface SearchResultResponse {
-  hasError: false;
-  searchResult: SearchResult;
-}
-type Response = ErrorResponse | SearchResultResponse;
-
-async function getData({searchParams = {}}: Props): Promise<Response> {
-  const {
-    publishers,
-    licenses,
-    query,
-    offset = '0',
-    sort = defaultSort,
-  } = searchParams;
-
-  const {sortBy, sortOrder} = sortMapping[sort] || {};
-
-  // Transform the string values from the query string to SearchOptions
-  const options = {
-    offset: +offset,
-    filters: {
-      publishers: publishers?.split(',').filter(id => !!id),
-      licenses: licenses?.split(',').filter(id => !!id),
-    },
-    sortBy: sortBy,
-    sortOrder: sortOrder,
-  };
-
-  if (!options.sortBy || !options.sortOrder || isNaN(options.offset)) {
-    return {
-      hasError: true,
-      errorMessage: 'Invalid options',
-      searchResult: null,
-    };
-  }
-
-  const validOptions: SearchOptions = options;
-
-  // Only add a search query if provided
-  if (query) {
-    validOptions.query = query;
-  }
+export default async function Home({searchParams}: Props) {
+  const {searchOptions, sortBy} = fromSearchParamsToSearchOptions(
+    searchParams ?? {}
+  );
 
+  let hasError, searchResult;
   try {
-    const searchResult = await datasetFetcher.search(validOptions);
-    return {searchResult, hasError: false};
-  } catch (error) {
-    return {hasError: true, errorMessage: 'Fetch error', searchResult: null};
+    searchResult = await datasetFetcher.search(searchOptions);
+  } catch {
+    hasError = true;
```

Comment:
Should we log the error? We're not using a logger at the moment (I need one soon in the backend code :)) but a `console.error` could be sufficient?